### ISSUE TITLE
[Bug]: launch-tae-multi-CN-tae-DN start-up kill query failed

### ISSUE THREAD

**Issue by heni02** [matrixorigin/matrixone#8028]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93):a142bd87396da73d417c7a2acaa964079e54440c
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

./mo-service -launch ./etc/launch-tae-multi-CN-tae-DN/launch.toml
![image](https://user-images.githubusercontent.com/113406637/218657337-d5205e97-2219-4245-a82c-9ce6096150e2.png)
![image](https://user-images.githubusercontent.com/113406637/218657274-88f3f29b-836c-4db9-a59b-bd4cd33706db.png)

**Comment by daviszhen** [comment]
we can not kill the query easily under current multi-cn environt.

**Comment by daviszhen** [comment]
can not be solved in 0.7

**Comment by daviszhen** [comment]
i am not focus on it today

**Comment by daviszhen** [comment]
i am not focus on it today

**Comment by daviszhen** [comment]
i am not focus on it today

**Comment by daviszhen** [comment]
i am not focus on it today

**Comment by daviszhen** [comment]
i am not focus on it today

**Comment by daviszhen** [comment]
I am not working on it

**Comment by daviszhen** [comment]
I am not working on it

**Comment by daviszhen** [comment]
I am not working on it

**Comment by daviszhen** [comment]
I am not working on it

**Comment by daviszhen** [comment]
I am not working on it

**Comment by daviszhen** [comment]
按照之前的说法，会与分布式一起讨论个方案

**Comment by daviszhen** [comment]
I am not working on it

**Comment by volgariver6** [comment]
I am working on it.

**Comment by volgariver6** [comment]
still working on it.

**Comment by volgariver6** [comment]
finish coding. Wait for the proxy code merge then commit this pr.

**Comment by volgariver6** [comment]
fixed by #8938. This feature only works with proxy module enabled.

**Comment by heni02** [comment]
Wait for the proxy implementation to return to this issue

**Comment by heni02** [comment]
connection to proxy ，also kill failed
commmit id :nightly-365ecb68 
![image](https://github.com/matrixorigin/matrixone/assets/113406637/48c68069-797e-4c79-bd82-092b20747985)
![image](https://github.com/matrixorigin/matrixone/assets/113406637/3beccf2c-3c41-4726-9328-1a86639253b5)
![image](https://github.com/matrixorigin/matrixone/assets/113406637/a4a2be7a-1bd7-4683-bb47-83be65fc36bf)

**Comment by volgariver6** [comment]
现在的单cn的环境也没办法kill query：

<img width="361" alt="image" src="https://github.com/matrixorigin/matrixone/assets/18366608/b112eeb5-5750-4790-adc8-db28b31038d2">

<img width="470" alt="image" src="https://github.com/matrixorigin/matrixone/assets/18366608/7bb632d2-a770-4cdb-8095-aec2bd398eae">

**Comment by volgariver6** [comment]
@daviszhen will handle it.

**Comment by daviszhen** [comment]
> 现在的单cn的环境也没办法kill query：
> 
> <img alt="image" width="361" src="https://user-images.githubusercontent.com/18366608/237560243-b112eeb5-5750-4790-adc8-db28b31038d2.png"> <img alt="image" width="470" src="https://user-images.githubusercontent.com/18366608/237560265-7bb632d2-a770-4cdb-8095-aec2bd398eae.png">
> commit id 是 [3ba01d4](https://github.com/matrixorigin/matrixone/commit/3ba01d47c1aaeaab8b11e61a1a925f683e1557e2)

单cn的问题已经在pr #9396 中修改了。

**Comment by daviszhen** [comment]
但是带proxy的多cn ctr+c还是有问题。
初步分析是给mysql client 的connection id 在proxy有问题。



![image](https://github.com/matrixorigin/matrixone/assets/60595215/b7c94c3a-e6fc-49d2-8c3b-4df43cd60fc0)

**Comment by daviszhen** [comment]
单机kill的pr #9396 已经merge了。

**Comment by volgariver6** [comment]
ctrl+C 的问题我再处理一下

**Comment by volgariver6** [comment]
fixed by #9448

**Comment by heni02** [comment]
confirm,closed
commit id:1db9350ffa37666a1ccc53ecc87cc75fba0aa290
./mo-service -launch etc/launch-with-proxy/launch.toml -with-proxy
<img width="774" alt="企业微信截图_500abe2b-9075-47bc-8403-12be4c8ef295" src="https://github.com/matrixorigin/matrixone/assets/113406637/5c7b16c5-5fbf-4448-aa06-0886d599f4b8">
![image](https://github.com/matrixorigin/matrixone/assets/113406637/fd38bd6d-5ac6-4a54-84b3-cd7c6200524f)
